cmd/snap-discard-ns/snap-discard-ns.c

```diff
@@ -40,6 +42,32 @@
 #define NSFS_MAGIC 0x6e736673
 #endif
 
+// Asserts the proces has sufficient capabilities being either run directly by root user or
+// through snap-confine.
+static void assert_caps(void) {
+    cap_t current SC_CLEANUP(cap_free) = cap_get_proc();
+
+    cap_value_t expected_caps[] = {
+        CAP_SYS_ADMIN,    /* umount */
+        CAP_DAC_OVERRIDE, /* for poking around /run/snapd */
+        CAP_CHOWN,        /* for lock file and directory */
+    };
+
+    for (size_t i = 0; i < SC_ARRAY_SIZE(expected_caps); i++) {
+        cap_value_t cap = expected_caps[i];
+        const char* cap_name SC_CLEANUP(cap_free) = cap_to_name(cap);
+
+        cap_flag_value_t set = CAP_CLEAR;
+        if (cap_get_flag(current, CAP_SYS_ADMIN, CAP_EFFECTIVE, &set) != 0) {
```

Comment:
This looks like a typo - I am guessing `CAP_SYS_ADMIN` be `cap` so we query each capability rather than repeatedly querying for `CAP_SYS_ADMIN`.

Reply:
Thanks for catching. I've pushed a fixup.